From: Luotao Fu <l.fu@pengutronix.de>
Date: Thu, 11 Dec 2008 11:49:56 +0100
Subject: [PATCH] fix depth calculation
Mainline status: accetped for 0.10.11 release

 depth should be a summary of the lengthes of all colors and the the alpha channel.

Signed-off-by: Luotao Fu <l.fu@pengutronix.de>
Acked-by: Wim Taymans <wim.taymans@chello.be>
---
 sys/fbdev/gstfbdevsink.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/sys/fbdev/gstfbdevsink.c b/sys/fbdev/gstfbdevsink.c
index 0fb7d98..c9a7e14 100644
--- a/sys/fbdev/gstfbdevsink.c
+++ b/sys/fbdev/gstfbdevsink.c
@@ -188,8 +188,10 @@ gst_fbdevsink_getcaps (GstBaseSink * bsink)
   caps = gst_caps_from_string (GST_VIDEO_CAPS_RGB_15);
   gst_caps_set_simple (caps,
       "bpp", G_TYPE_INT, fbdevsink->varinfo.bits_per_pixel,
-      "depth", G_TYPE_INT, fbdevsink->varinfo.bits_per_pixel
-      - fbdevsink->varinfo.transp.length,
+      "depth", G_TYPE_INT, fbdevsink->varinfo.red.length +
+	fbdevsink->varinfo.green.length +
+	fbdevsink->varinfo.blue.length +
+        fbdevsink->varinfo.transp.length,
       "endianness", G_TYPE_INT, endianness,
       "red_mask", G_TYPE_INT, rmask,
       "green_mask", G_TYPE_INT, gmask, "blue_mask", G_TYPE_INT, bmask, NULL);
-- 
1.5.6.5

